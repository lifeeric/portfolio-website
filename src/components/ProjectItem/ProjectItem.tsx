import * as React from "react"
import { FiExternalLink } from "react-icons/fi"
import { FaGithub } from "react-icons/fa"
import { ProjectImg } from "./ProjectImg/ProjectImg"

import "./ProjectItem.scss"

interface Props {
  name: string
  git: string
  live: string
  image: string
}

export const ProjectItem: React.FC<Props> = ({ image, name, live, git }) => {
  return (
    <div className="project__box">
      {/* {console.log(image, " => item")} */}
      <ProjectImg src={image} />
      <div className="project__body">
        <p className="project__header">{name}</p>
        <div className="project__source">
          <a target="_blank" href={git}>
            <FaGithub />
          </a>
          <a target="_blank" href={live}>
            <FiExternalLink />
          </a>
        </div>
      </div>
    </div>
  )
}
