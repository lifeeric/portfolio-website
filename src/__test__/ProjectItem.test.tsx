import * as React from "react"
import { shallow, mount } from "enzyme"

import { ProjectItem } from "../components/ProjectItem/ProjectItem"

describe("<Header /> testing", () => {
  let wrapper: any
  const data = {
    name: "me",
    git: "@lifeeric",
    live: "empty",
    image: "jpg",
  }
  beforeEach(() => {
    wrapper = shallow(<ProjectItem {...data} />)
  })

  it("should render well", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should give the value", () => {
    console.log(wrapper.find(".project__header").text())
    expect(wrapper.find(".project__header").text()).toBe("me")
  })
})
