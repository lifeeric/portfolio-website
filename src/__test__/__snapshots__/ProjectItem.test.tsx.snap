// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> testing should render well 1`] = `
<div
  className="project__box"
>
  <Component
    image="jpg"
  />
  <div
    className="project__body"
  >
    <p
      className="project__header"
    >
      me
    </p>
    <div
      className="project__source"
    >
      <a
        href="@lifeeric"
      >
        <FaGithub />
      </a>
      <a
        href="empty"
      >
        <FiExternalLink />
      </a>
    </div>
  </div>
</div>
`;
